libraries/bot-azure/src/main/java/com/microsoft/bot/azure/blobs/BlobsTranscriptStore.java

```diff
@@ -0,0 +1,423 @@
+                       ChannelAccount recipient = new ChannelAccount() {
+                           {
+                               setId("deleted");
+                               setRole(activityAndBlob.getLeft().getRecipient().getRole());
+                           }
+                       };
+                       // tombstone the original message
+                       Activity tombstonedActivity = new Activity() {
+                           {
+                               setType(ActivityTypes.MESSAGE_DELETE);
+                               setId(activityAndBlob.getLeft().getId());
+                               setFrom(from);
+                               setRecipient(recipient);
+                               setLocale(activityAndBlob.getLeft().getLocale());
+                               setLocalTimestamp(activityAndBlob.getLeft().getTimestamp());
+                               setTimestamp(activityAndBlob.getLeft().getTimestamp());
+                               setChannelId(activityAndBlob.getLeft().getChannelId());
+                               setConversation(activityAndBlob.getLeft().getConversation());
+                               setServiceUrl(activityAndBlob.getLeft().getServiceUrl());
+                               setReplyToId(activityAndBlob.getLeft().getReplyToId());
+                           }
+                       };
+
+                       logActivityToBlobClient(tombstonedActivity, activityAndBlob.getRight(), true).join();
+                   }
+                });
+
+                return CompletableFuture.completedFuture(null);
+            default:
+                String blobName = this.getBlobName(activity);
+                BlobClient blobClient = containerClient.getBlobClient(blobName);
+                logActivityToBlobClient(activity, blobClient, true).join();
+                return CompletableFuture.completedFuture(null);
+        }
+    }
+
+    public CompletableFuture<PagedResult<Activity>> getTranscriptActivities(String channelId, String conversationId,
+                                                                            @Nullable String continuationToken,
+                                                                            OffsetDateTime startDate) {
+        if (startDate == null) {
+            startDate = OffsetDateTime.now(ZoneId.of("UTC"));
+        }
+
+        final int pageSize = 20;
+
+        if (StringUtils.isBlank(channelId)) {
+            throw new IllegalArgumentException("Missing channelId");
+        }
+
+        if (StringUtils.isBlank(conversationId)) {
+            throw new IllegalArgumentException("Missing conversationId");
+        }
+
+        PagedResult<Activity> pagedResult = new PagedResult<Activity>();
+
+        String token = null;
+        List<BlobItem> blobs = new ArrayList<BlobItem>();
+        do {
+            String prefix = String.format("%1$s/%2$s/", sanitizeKey(channelId), sanitizeKey(conversationId));
```

Comment:
```suggestion
            String prefix = String.format("%s/%s/", sanitizeKey(channelId), sanitizeKey(conversationId));
```

As we are using the index by order, not necessary to specify the index